src/averageAges.js

```diff
@@ -55,7 +67,22 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = people
+    .filter(child => people
+      .some(woman => woman.name === child.mother));
+
+  const son = people
+    .filter(child => people
+      .some(woman => child.mother === woman.name && child.sex === 'm'));
```

Comment:
[checklist point 2](https://github.com/mate-academy/js_average-ages/blob/master/checklist.md)

Better rename to `sons` because there stored more then 1 son.